src/Core/src/Handlers/Layout/LayoutExtensions.cs

```diff
@@ -9,14 +9,34 @@ internal static class LayoutExtensions
 
 		public static int GetLayoutHandlerIndex(this ILayout layout, IView view)
```

Comment:
Maybe someone on the MAUI team can comment: do we have enough unit tests that check this method would return the right results?

Reply:
For reference: https://github.com/dotnet/maui/blob/74c8c146957ccaf9be02396365c2cc0617ba832d/src/Core/tests/UnitTests/Layouts/ZIndexTests.cs#L171-L202

We may want to add
```cs
		[Fact]
		public void LayoutHandlerIndexIsNegativeWhenChildIsNotFound()
		{
			var layout = new FakeLayout();
			var view0 = CreateTestView(zIndex: 0);

			Assert.Equal(-1, layout.GetLayoutHandlerIndex(view0));
		}

		[Fact]
		public void LayoutHandlerIndexEscalatesToChildIndexWhenZIndexIsEqual()
		{
			var layout = new FakeLayout();
			var view0 = CreateTestView(zIndex: 0);
			var view1 = CreateTestView(zIndex: 10);
			var view2 = CreateTestView(zIndex: 0);
			layout.Add(view0);
			layout.Add(view1);
			layout.Add(view2);

			Assert.Equal(0, layout.GetLayoutHandlerIndex(view0));
			Assert.Equal(2, layout.GetLayoutHandlerIndex(view1));
			Assert.Equal(1, layout.GetLayoutHandlerIndex(view2));
		}
```

Reply:
If this test will cover cases that are not covered, then we should add it. Better to have too many tests (within reason - but we are not there yet) than not enough.

Reply:
Yeah, @albyrock87 go ahead and add these. Otherwise this one LGTM? @mattleibow is there someone more knowledgable about layout that would also make sense to review?

Reply:
@mattleibow @jonathanpeppers tests have been added

```
        LayoutHandlerIndexIsNegativeWhenChildIsNotFound
        LayoutHandlerIndexPreservesAddOrderForEqualZIndexes
```

I've also removed the reflection thanks to 
```
[assembly: InternalsVisibleTo("Core.Benchmarks")]
```
which seemed appropriate.